### PR TITLE
[full-ci][tests-only] add CI pipeline for checking lint issues using gherlint

### PR DESCRIPTION
**Pull request by prashant-gurung899**
## Description
This PR adds new CI pipeline to check for lint issues using gherlint.

## Related Issue

- Related to : https://github.com/owncloud/QA/issues/833

## How Has This Been Tested?

- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by prashant-gurung899** [comment]
The failure that is supposed to happen when there are any lint issues in feature files is here: https://drone.owncloud.com/owncloud/ocis/36609/2/2

File: Makefile [hunk 1]
```diff
@@ -235,6 +235,10 @@ golangci-lint-fix:
         $(MAKE) --no-print-directory -C $$mod golangci-lint-fix; \
     done
 
+.PHONY: gherkin-features-lint
+gherkin-features-lint:
```
Comment (saw-jan) [inline Makefile:239]:
```suggestion
test-gherkin-lint:
```

File: .drone.star [hunk 2]
```diff
@@ -724,6 +725,28 @@ def checkTestSuitesInExpectedFailures(ctx):
         },
     }]
 
+def checkLintIssuesWithGherlint(ctx):
```
Comment (saw-jan) [inline .drone.star:728]:
```suggestion
def checkGherkinLint(ctx):
```

File: .drone.star [hunk 3]
```diff
@@ -724,6 +725,28 @@ def checkTestSuitesInExpectedFailures(ctx):
         },
     }]
 
+def checkGherkinLint(ctx):
+    return [{
+        "kind": "pipeline",
+        "type": "docker",
+        "name": "check-lint-issues-with-gherlint",
```
Comment (saw-jan) [inline .drone.star:732]:
```suggestion
        "name": "check-gherkin-standard",
```

File: Makefile [hunk 4]
```diff
@@ -235,6 +235,10 @@ golangci-lint-fix:
         $(MAKE) --no-print-directory -C $$mod golangci-lint-fix; \
     done
 
+.PHONY: test-gherkin-lint
+test-gherkin-lint:
+	gherlint tests/acceptance/features -c tests/acceptance/config/.gherlintrc.json
+
```
Comment (saw-jan) [inline Makefile:241]: we could also add `test-gherkin-lint-fix` command

Comment (prashant-gurung899) [reply Makefile:241]: Sure.I'm on it.

File: .drone.star [hunk 5]
```diff
@@ -724,6 +725,28 @@ def checkTestSuitesInExpectedFailures(ctx):
...
+        "kind": "pipeline",
+        "type": "docker",
+        "name": "check-gherkin-standard",
+        "steps": [
+            {
+                "name": "lint-feature-files",
+                "image": OC_CI_NODEJS % DEFAULT_NODEJS_VERSION,
+                "commands": [
+                    "npm install -g @gherlint/gherlint@1.1.0",
+                    "gherlint -c tests/acceptance/config/.gherlintrc.json tests/acceptance/features",
```
Comment (SagarGi) [inline .drone.star:739]: i see we have make command for it. cant we just use it instead of this? @saw-jan @prashant-gurung899

Comment (prashant-gurung899) [reply .drone.star:739]: I think should install the package anyhow before using it in drone. I'm not sure. @saw-jan ?

**Review by saw-jan — APPROVED** [review]
LGTM :+1:

**Review by phil-davis — APPROVED** [review]

**Review by amrita-shrestha — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]

**Pull request merged**